DOCSP-24556: issues & help page (#18)
mongodb/docs-csharp: source/issues-and-help.txt

=============
Issues & Help
=============

.. default-domain:: mongodb

We are lucky to have a vibrant community of {+driver-long+}
users. We recommend seeking support for general
questions through the `MongoDB Community Forums
<https://www.mongodb.com/community/forums>`__.

Bugs / Feature Requests
-----------------------

To report a bug or to request a new feature in the {+driver-short+},
please open an issue in JIRA, our issue management tool, using the
following steps:

1. `Create a JIRA account <https://jira.mongodb.org>`__.
#. Navigate to `the CSHARP Driver project <https://jira.mongodb.org/browse/CSHARP>`_.
#. Click :guilabel:`Create Issue`. Please provide as much information as possible about the
   issue and the steps to reproduce it.

Bug reports in JIRA for the CSHARP Driver project can be viewed by everyone.

If you identify a security vulnerability in the driver or in any other
MongoDB project, please report it according to the instructions found in
:manual:`Create a Vulnerability Report </tutorial/create-a-vulnerability-report>`.

Pull Requests
-------------

> We are happy to accept contributions to help improve the driver. We will
> guide user contributions to ensure they meet the standards of the 
> codebase.
> 
> To contribute to the official driver, follow the `development and testing guidelines
> <https://github.com/mongodb/mongo-csharp-driver/blob/master/CONTRIBUTING.md>`__
> to ensure your code meets the quality standards.

Reviewer: Suggestion: 
Rearrange the sentences so they're chronological. That could also help reduce repetition.

```suggestion
We are happy to accept contributions to help improve the driver.
To contribute to the official driver, follow the `development and testing guidelines
<https://github.com/mongodb/mongo-csharp-driver/blob/master/CONTRIBUTING.md>`__
We will guide user contributions to ensure they meet the standards of the 
codebase.
```

=============
Issues & Help
=============

.. default-domain:: mongodb

> We are lucky to have a vibrant community of MongoDB .NET/C# Driver

Reviewer: Per [this prior comment](https://github.com/mongodb/docs-csharp/pull/18#discussion_r980103407), use the source constant. Let me know if this doesn't make sense and we can discuss.

```suggestion
We are lucky to have a vibrant community of {+driver-long+}
```

=============
Issues & Help
=============

.. default-domain:: mongodb

We are lucky to have a vibrant MongoDB C#/.NET community

> that includes users with varying levels of experience using
> the driver. We find the quickest way to get support for general
> questions is through the `MongoDB Community Forums

Reviewer: Suggestion:

In the original version, "We are lucky to have" refers to the community that specifically has varying levels of experience, but it's not clear why the users with varying levels of experience makes MongoDB lucky or how it qualifies the users of the forums as the quickest way to get support. Instead, consider omitting or explaining the connection.
 
I see this is essentially what is written on the Manual "Technical Support" page, but that doesn't mean it's clear!

```suggestion
of users. We recommend seeking support for general
questions through the `MongoDB Community Forums
```

<https://www.mongodb.com/community/forums>`__.

> To learn more, refer to our :manual:`support channels </support>`.

Reviewer: Suggestion:

A couple items here:
1) When cross-linking, make sure it's specific about what page it's linking to (see [Links and Cross Reference](https://www.mongodb.com/docs/meta/style-guide/style/links-and-cross-references/#link-and-cross-reference-text)s in the style guide) 
"To learn more about MongoDB support options..."

2) The Technical Support page links to the forums and general, non-driver specific MongoDB resources. The intro text for the forums is duplicated which is a good indicator there's redundant text. One solution could be to omit this link (what I would suggest) and another could be to focus on describing the Technical Support page instead of repeating the content of one of its sections.

Author: chose to omit the link. I see that the technical support page links right back to the community forums.

Bugs / Feature Requests
-----------------------

> To report a bug or to request a new feature in the C#/.NET driver,

Reviewer: Suggestion:

Since this repo uses [this source constant](https://github.com/mongodb/docs-csharp/blob/master/snooty.toml#L18) for the name of the driver, it is helpful to use in every instance. Otherwise, if the name ever changes, someone will need to search through the entire set of docs to find and replace rather than changing the source constant only.

Author: I understood this comment to mean that this page should follow the order of .NET/C# where mentioned.

> please open a case in our issue management tool, JIRA:

Reviewer: ```suggestion
please open an issue in JIRA, an issue management tool, using the following steps:
```

> * `Create an account and login <https://jira.mongodb.org>`_.

Reviewer: I think what you are saying is "log in" after creating an account rather than creating a login. Or if logging in is automatic, I don't think you need to list it.

A couple alternatives:

```suggestion
* `Create an account and log in <https://jira.mongodb.org>`_.
```

```suggestion
* `Create a JIRA account <https://jira.mongodb.org>`_.
```

* Navigate to `the .NET Driver project <https://jira.mongodb.org/browse/CSHARP>`_.

> * Click **Create Issue**. Please provide as much information as possible about the

Reviewer: ```suggestion
* Click :guilabel:`Create Issue`. Please provide as much information as possible about the
```

> * Click **Create Issue**. Please provide as much information as possible about the

Reviewer: Suggestion: I think it works best to use an ordered list to represent steps that should be performed in order.
https://www.mongodb.com/docs/meta/style-guide/style/lists/#lists

  issue and the steps to reproduce it.

> Bug reports in JIRA for the C#/.NET driver and the Core Server (i.e. SERVER) project are **public**.

Reviewer: Suggestion:

I think it could be helpful to explain what "public" means in this context for anyone that might not be familiar. Also, not sure if the user might be confused by specifying the SERVER project here since most projects in our JIRA are public.

```suggestion
Bug reports in JIRA for the C#/.NET driver project can be viewed by everyone.
```

> If you’ve identified a security vulnerability in a driver or any other

Reviewer: Suggestion: 
Prefer simple present tense when possible.

```suggestion
If you identify a security vulnerability in the driver or in any other
```

MongoDB project, please report it according to the instructions found in

> the :manual:`Create a Vulnerability Report </tutorial/create-a-vulnerability-report>`.

Reviewer: ```suggestion
:manual:`Create a Vulnerability Report </tutorial/create-a-vulnerability-report>`.
```

Pull Requests
-------------

We are happy to accept contributions to help improve the driver. We will
guide user contributions to ensure they meet the standards of the
codebase. Please ensure that any pull requests include documentation,

> tests, and pass the existing test suite.

Reviewer: Question: I wasn't able to find the existing test suite mentioned here. The Contributing.md file linked in this section only mentions writing your own tests. Are there directions we can link to in order to pass the existing test suite? Or should this description be changed to match the requirements on Contributing.md?

Author: I used the wording from the outline document... but yes, it appears that there are no given tests. I updated the language.

Reviewer: Did the DBX team provide information on how they expect to receive contributions? If you didn't get that text from them whether directly, if that's not what the DBX team wants to communicate, definitely update whatever is needed (Contributing.md and/or this paragraph) to make it accurate.
